src/features/users/usersSlice.ts

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable no-param-reassign */
+import {
+  // PayloadAction,
+  createAsyncThunk,
+  createSlice,
+} from '@reduxjs/toolkit';
+import { User } from '../../types/User';
+import { getUsers } from '../../api/users';
+
+type UsersState = {
+  users: User[],
+  loading: boolean,
+  error: string,
+};
+
+const initialState: UsersState = {
+  users: [],
+  loading: false,
+  error: '',
+};
+
+export const init = createAsyncThunk('users/fetchUsers', async () => {
+  const usersFromServer = await getUsers();
+
+  return usersFromServer;
+});
+
+export const usersSlice = createSlice({
+  name: 'users',
+  initialState,
+  reducers: {},
+
+  extraReducers: (builder) => {
+    builder.addCase(init.pending, (state) => {
+      state.loading = true;
+    });
+
+    builder.addCase(init.fulfilled, (state, action) => {
+      state.users = action.payload;
+      state.loading = false;
+    });
+
+    builder.addCase(init.rejected, (state) => {
+      state.error = 'Error';
```

Comment:
Instead of hardcoding 'Error', consider using a more descriptive message or the error message returned by the rejected promise.

Reply:
fixed